src/constructs/autoscaling/asg.ts

```diff
@@ -101,10 +105,11 @@ export class GuAutoScalingGroup extends AutoScalingGroup {
 
       // Do not use the default AWS security group which allows egress on any port.
       // Favour HTTPS only egress rules by default.
-      securityGroup: GuHttpsEgressSecurityGroup.forVpc(scope, props),
+      securityGroup: GuHttpsEgressSecurityGroup.forVpc(scope, { app: props.app, vpc: props.vpc }),
```

Comment:
Passing in `props` meant `existingLogicalId` was copied to the `GuHttpsEgressSecurityGroup` too. When it came time to synth the stack, it failed because two resources were being created with the same logicalId (the `GuHttpsEgressSecurityGroup` and the `GuAutoScalingGroup`).

This took the best part of two days to diagnose 😨 !

Reply:
😱

Reply:
https://github.com/microsoft/TypeScript/issues/12936